tests/test_global.py

```diff
@@ -0,0 +1,292 @@
+        with open(local_tiff_path, "wb") as file:
+            file.write(response.content)
+
+        print(f"zip file has been downloaded to {local_tiff_path}")
+    else:
+        print("Failed to download zip file:", response.status_code)
+
+
+georegion = nso.nso_georegion(
+    path_to_geojson=path_geojson,
+    output_folder=output_path,
+    username=nso_username,
+    password=nso_password,
+)
+links = georegion.retrieve_download_links(max_diff=0.5, start_date="2011-01-01")
+
+
+### Check if test data exists else download the data.
+
+if not os.path.exists(
+    output_path + "/20230513_104139_PNEO-03_1_1_30cm_RGBNED_12bit_PNEO.zip"
+):
+    download_zip(
+        "https://e34a505986aa74678a5a0e0f.blob.core.windows.net/satellite-images-nso/Test_regions/20230513_104139_PNEO-03_1_1_30cm_RGBNED_12bit_PNEO.zip"
+    )
+
+if not os.path.exists(
+    output_path + "/20191202_110523_SV1-04_SV_RD_11bit_RGBI_50cm.zip"
+):
+    download_zip(
+        "https://e34a505986aa74678a5a0e0f.blob.core.windows.net/satellite-images-nso/Test_regions/20230513_104139_PNEO-03_1_1_30cm_RGBNED_12bit_PNEO.zip"
+    )
+
+
+### Link tests.
+
+
+def test_found_links():
+
+    assert len(links) > 0, "No links found error!"
+
+
+def test_link_columns():
+    # Columns to check for existence
+    columns_to_check = [
+        "link",
+        "percentage_geojson",
+        "missing_polygon",
+        "covered_polygon",
+    ]
+
+    # Check if columns exist
+    existing_columns = [col for col in columns_to_check if col in links.columns]
+
+    assert len(existing_columns) == len(
+        columns_to_check
+    ), "Certain columns are not in the links"
+
+
+def test_pneo_links():
+    # Test if PNEO links are found
+    pneo_links = links[links["resolution"] == "30cm"]
+    pneo_links = pneo_links[pneo_links["link"].str.contains("RGBNED")]
+    pneo_links = pneo_links.sort_values("percentage_geojson")
+
+    assert len(pneo_links) > 0, "No PNEO links found error!"
+
+
+def test_superview_links():
+    # Example filter on resolution and bands
+    superview_links = links[links["resolution"] == "50cm"]
+    superview_links = superview_links[superview_links["link"].str.contains("RGBI")]
+    superview_links = superview_links.sort_values("percentage_geojson")
+
+    assert len(superview_links) > 0, "No  PNEO links found error!"
+
+
+# Cropping tests
+
+
+def pneo_polygon_crop():
+
+    georegion = nso.nso_georegion(
+        path_to_geojson="C:/repos/satellite-images-nso/input_data/Test_region.geojson",
+        output_folder=output_path,
+        username=nso_username,
+        password=nso_password,
+    )
+
+    filepath = georegion.execute_link(
+        "https://api.satellietdataportaal.nl/v1/download/30cm_RGBNED_12bit_PNEO/20230513_104139_PNEO-03_1_1",
+        plot=False,
+    )
+
+    assert filepath, "No file has been downloaded or extracted for a polygon in PNEO!"
+
+    # Remove the file for later tests
+    os.remove(filepath)
```

Comment:
So if your assert statement fails, this os.remove line of code will never be reached. If you want to always run some code after your test, you could for instance use fixtures:

https://stackoverflow.com/questions/22627659/run-code-before-and-after-each-test-in-py-test